src/eth_provider/provider.rs

```diff
@@ -635,7 +635,16 @@ where
         let calldata: Vec<FieldElement> = data.into_iter().map_into().collect();
 
         let gas_limit = into_via_try_wrapper!(request.gas.unwrap_or_else(|| U256::from(CALL_REQUEST_GAS_LIMIT)));
-        let gas_price = into_via_try_wrapper!(request.gas_price.unwrap_or_default());
+
+        // We cannot unwrap_or_default() here because Kakarot.eth_call will
+        // Reject transactions with gas_price < Kakarot.base_fee
+        let gas_price = {
```

Comment:
⚠️ boyscout commit:
- currently on Kakarot sepolia, if base_fee of the network > 0 and tx's gas_price == 0, eth_call will fail and user will be confused